### DISCUSSION TITLE
Image/Markdown previewer not working

### DISCUSSION THREAD

**Discussion by pwwang** [coder/code-server#4200] [Q&A]
Originally posted at: https://github.com/cdr/code-server/issues/3826
Related: https://github.com/microsoft/vscode/issues/98768#issuecomment-636404934

I upgraded to 3.12, the image previewer still fails:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'addEventListener')
    at main.js:399
main.js:203 Uncaught (in promise) Error: Service Workers are not enabled. Webviews will not work. Try disabling private/incognito mode.
    at main.js:203
    at new Promise (<anonymous>)
    at main.js:201
```

Then I set the safe browsing to no protection:

```
vscode.browserified.js:129 error [Service Worker] navigator is undefined
webWorkerExtensionHost.ts:143 The web worker extension host is started without an iframe sandbox!
start @ webWorkerExtensionHost.ts:143
abstractExtensionService.ts:1081 Extension 'ms-python.python' uses PROPOSED API which is subject to change and removal without notice.
updateEnableProposedApi @ abstractExtensionService.ts:1081
abstractExtensionService.ts:1081 Extension 'ms-toolsai.jupyter' uses PROPOSED API which is subject to change and removal without notice.
updateEnableProposedApi @ abstractExtensionService.ts:1081
log.ts:301   ERR a text search provider for the scheme 'file' is already registered: Error: a text search provider for the scheme 'file' is already registered
```

I am not sure how to get the service workers working for me.

**Comment by jsjoeio** [comment]
Hmm...It doesn't seem like this is going to work unless you have Service Workers enabled. How are you using code-server? And are you using HTTPS?

**Reply by pwwang** [reply]
Just untar the release and run from `bin/code-server`, no https.

**Reply by jsjoeio** [reply]
Hmm... and I assume you're following the installation instructions [here](https://coder.com/docs/code-server/v3.12.0/install)?

Is it running on localhost on your local machine or is this a remote machine?

**Reply by pwwang** [reply]
I followed this exactly: https://coder.com/docs/code-server/v3.12.0/install#standalone-releases

It's running on a remote machine but accessed via ssh forwarded port.

**Reply by nooobcoder** [reply]
@jsjoeio I faced this issue with WebViews in my hosted instance, as I was very puzzled to implement HTTPS. So, I created a different chrome profile by ignoring cert errors (on linux) using this command, this ignores cert errors and enables Service workers without issues.
@pwwang have a look here, isn't it similar to your issue? #3757 
`google-chrome-beta --ignore-certificate-errors --user-data-dir=/home/ankurpaul/ChromeForCodeServer  & exit `

**Reply by jsjoeio** [reply]
^Give that a try and let us know if it works!

(Thanks @nooobcoder for popping in and sharing that!)
